controllers/drplacementcontrol_controller.go

```diff
@@ -613,7 +615,15 @@ func (r *DRPlacementControlReconciler) Reconcile(ctx context.Context, req ctrl.R
 	if isBeingDeleted(drpc, placementObj) {
 		// DPRC depends on User PlacementRule/Placement. If DRPC or/and the User PlacementRule is deleted,
 		// then the DRPC should be deleted as well. The least we should do here is to clean up DPRC.
-		return r.processDeletion(ctx, drpc, placementObj, logger)
+		err := r.processDeletion(ctx, drpc, placementObj, logger)
+		if err != nil {
+			// update drpc progression only on err
+			logger.Info(fmt.Sprintf("Error in deleting DRPC: (%v)", err))
+
+			return r.setProgressionAndUpdate(ctx, drpc, rmn.ProgressionDeleting)
```

Comment:
This function has nothing to do with `ctrl.Result{}`. I would make it return `nil` or `error`. That's all.  So in this line, I would change it to this:
```
return ctrl.Result{}, r.setProgressionAndUpdate(ctx, drpc, rmn.ProgressionDeleting)
```